### ISSUE TITLE
cordova ios 5.1.1 + wkwebview + angularjs  - got blank screen when app is loaded

### ISSUE THREAD

**Issue by yuanzuo** [apache/cordova-ios#835]
# Bug Report
cordova ios 5.1.1 wkwebview blank screen 
## Problem
when app is loaded in xcode, screen is blank
### What is expected to happen?
screen should be displayed

### What does actually happen?
I see this error in Xcode:
2020-05-09 11:41:50.874859-0400 <my app name>[31815:320312] ERROR: Error: [$compile:tpload] http://errors.angularjs.org/1.3.8/$compile/tpload?p0=templates%2FHome.html

### Environment, Platform, Device

My environment is below:

cordova: 9.0.0
cordova-ios: 5.1.1
installed cordova-plugin-wkwebview-engine (version 1.2.1)

added the following in the config.xml:
<feature name="CDVWKWebViewEngine">
<param name="ios-package" value="CDVWKWebViewEngine"/>
</feature>
<preference name="CordovaWebViewEngine" value="CDVWKWebViewEngine"/>
<preference name="WKWebViewOnly" value="true"/>

## Checklist

- [ ] I searched for existing GitHub issues
- [ ] I updated all Cordova tooling to most recent version
- [ ] I included all the necessary information above

**Comment by yuanzuo** [comment]
added the following in the config.xml:
<feature name="CDVWKWebViewEngine">
<param name="ios-package" value="CDVWKWebViewEngine"/>
</feature>
<preference name="CordovaWebViewEngine" value="CDVWKWebViewEngine"/>
<preference name="WKWebViewOnly" value="true"/>

**Comment by yuanzuo** [comment]
added the following in the config.xml:
sdgsfgsdg
sgdsdg

**Comment by breautek** [comment]
As the angular error suggests, it's failing to load the template. You should check your:
- template paths
- Don't forget that your app is hosted on the filesystem, not on a webserver, so you have to use relative URLs. Urls will be relative to your html document.
- All your web assets should be inside the `<cordova-project>/www` directory.

If you need further assistance, I invite you to our [slack](http://slack.cordova.io/) channel, which is a more appropriate place for getting support. If you think you've found a bug, then please open a new ticket and make sure you include code and other relevant information for reproduction.

**Comment by subbuy** [comment]
With same versions I ran into same issue with same error message. What was the resolution for closing the bug?

**Comment by breautek** [comment]
> What was the resolution for closing the bug?

It's not a Cordova bug. I'll elaborate, and provide some potential workarounds.

The OP didn't provide any information that provide the actual cause of this particular issue, so to properly debug this, you would have to look at the safari web inspector to see what JS errors are occurring. Do note that safari doesn't capture log events until only after the inspector is opened/connected to the app. This means you'll need to refresh the webview after you connected the web inspector. You can do this by pressing cmd + r while having the safari web inspector focused.

If you're getting an error that suggest that template file cannot be found, then what I said back in [May](https://github.com/apache/cordova-ios/issues/835#issuecomment-626246151) applies.

Additionally, another common issue with angular apps specifically is the fact that they make heavy use of `XMLHttpRequest` for loading template files and other assets. On cordova 5.x with using WKWebView, this isn't supported out of the box due to the browser enforcing CORs. So if you're receiving a CORs issue you have 2 options.

1. Upgrade to cordova-ios@6 and use schemes as [documented here](https://cordova.apache.org/howto/2020/07/18/uiwebview-warning.html). Do note that using schemes will change your origin, so if you're updating an existing app that is released, your clients will lose access to their existing web storage data (indexed DB, cookies, local storage, etc).

2. If you cannot afford to lose access to the web storage, then staying on cordova-ios@5.1.1 is probably your best bet. You'll need to use a plugin such as [cordova-plugin-wkwebview-file-xhr](https://github.com/oracle/cordova-plugin-wkwebview-file-xhr) to workaround the CORS enforcement on the file protocol. Do note that the plugin doesn't currently support 6.x yet, tracking issue can be found at https://github.com/oracle/cordova-plugin-wkwebview-file-xhr/issues/55

**Comment by Masterxiechengxun** [comment]
You can try plugin [cordova-plugin-wkwebview-file-xhr](https://github.com/oracle/cordova-plugin-wkwebview-file-xhr)
